codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java

```diff
@@ -131,11 +154,32 @@ public HttpHeaders clear() {
         }
 
         @Override
+        public Entry<CharSequence, CharSequence> forEachEntry(EntryVisitor<CharSequence> visitor) throws Exception {
+            return null; // Since this is an empty header collection
+        }
+
+        @Override
         public Iterator<Entry<String, String>> iterator() {
             return entries().iterator();
         }
     };
 
+    private final EntryVisitor<CharSequence> setAllVisitor = new EntryVisitor<CharSequence>() {
```

Comment:
Thanks for being consistent :).  This has lead me to re-review the DefaultHeaders implementation and realize that this member is a good candidate for lazy initialization.  See PR https://github.com/netty/netty/pull/3342.  If you feel this is a good improvement then lets follow the lazy initialization approach.